eng/native/configurecompiler.cmake

```diff
@@ -1,3 +1,10 @@
+# Due to how we build the libraries native build as part of the CoreCLR build as well as standalone,
+# we can end up coming to this file twice. Only run it once to simplify our build.
+if (CLR_CMAKE_COMPILER_CONFIGURED)
+  return()
+endif()
+set(CLR_CMAKE_COMPILER_CONFIGURED ON)
```

Comment:
We can use the global [`include_guard`](https://cmake.org/cmake/help/latest/command/include_guard.html) for this in cmake 3.10+.